lib/msal-node/src/client/ConfidentialClientApplication.ts

```diff
@@ -97,6 +97,15 @@ export class ConfidentialClientApplication extends ClientApplication implements
             clientAssertion
         };
 
+        /*
+         * valid request should not have "common" or "organization" in lieu of the tenant_id in the authority in the auth configuration
+         * example authority: "https://login.microsoftonline.com/TenantId",
+         */
+        const tenantId = validRequest.authority.split("/")[3];
```

Comment:
For reference, the .NET algorithm is: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs#L367

Reply:
All this does is remove the "/"s from the authority URL by splitting the string at each occurrence of it. In this case, we want the very last chunk of string. Perhaps I could change this to: 

```
const strings = validRequest.authority.split("/");
const tenantId = strings[strings.length - 1];
```

it appears to me that this is similar to what .NET does.

Reply:
I don't see anything in the common or node utils that does this already.

Reply:
`Authority.ts` has this code:
```
    /**
     * Get tenant for authority.
     */
    public get tenant(): string {
        return this.canonicalAuthorityUrlComponents.PathSegments[0];
    }
```

can we reuse it?

Reply:
@sameerag I can't reuse that method because it's part of the Authority class, and here we only have an authority string. However, that code uses the method getUrlComponents from [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/src/url/UrlString.ts#L147). I've basically rewritten that functionality in the comment I left above. We don't need all the urlComponents (Protocol, HostNameAndPort, AbsolutePath, QueryString as an IURI, so I think what I wrote in my comment above should suffice.